yogstation/code/game/gamemodes/gangs/smugglerbeacon.dm

```diff
@@ -0,0 +1,137 @@
+/*
+	Gang stashbox
+	Money can be inserted into it for gang to use
+	but money cannot be taken out directly
+	To widthdraw money, one has to use the gangtool
+	Destroying the stashbox will drop a significant
+	portion of the money inside, but not all of it
+*/
+/obj/item/stashbox
+	name = "stashbox"
+	desc = "a secure stash box criminals use to hide their money"
+	icon = 'icons/obj/module.dmi'
+	icon_state = "depositbox"
+	w_class = WEIGHT_CLASS_NORMAL
+	level = 1 // Makes the item hide under floor tiles
+	var/datum/bank_account/registered_account
+
+/obj/item/stashbox/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/holochip))
+		insert_money(W, user)
+		return
+	else if(istype(W, /obj/item/stack/spacecash))
+		insert_money(W, user, TRUE)
+		return
+	else if(istype(W, /obj/item/coin))
+		insert_money(W, user, TRUE)
+		return
+	else
+		return ..()
+
+/obj/item/stashbox/hide(intact)
+	if(intact)
+		invisibility = INVISIBILITY_OBSERVER
+		anchored = TRUE //otherwise you can start pulling, cover it, and drag around an invisible backpack.
+		icon_state = "[initial(icon_state)]2"
+		ADD_TRAIT(src, TRAIT_T_RAY_VISIBLE, TRAIT_GENERIC)
+	else
+		invisibility = initial(invisibility)
+		anchored = FALSE
+		icon_state = initial(icon_state)
+		REMOVE_TRAIT(src, TRAIT_T_RAY_VISIBLE, TRAIT_GENERIC)
+
+/obj/item/stashbox/proc/insert_money(obj/item/I, mob/user, physical_currency)
+	var/cash_money = I.get_item_credit_value()
+	if(!cash_money)
+		to_chat(user, "<span class='warning'>[I] doesn't seem to be worth anything!</span>")
+		return
+
+	if(!registered_account) // Shouldn't happen but lets leave it in just in case
+		to_chat(user, "<span class='warning'>[src] doesn't have a linked account to deposit [I] into!</span>")
+		return
+
+	registered_account.adjust_money(cash_money)
+	if(physical_currency)
+		to_chat(user, "<span class='notice'>You stuff [I] into [src]. It disappears in a small puff of bluespace smoke, adding [cash_money] credits to the linked account.</span>")
+	else
+		to_chat(user, "<span class='notice'>You insert [I] into [src], adding [cash_money] credits to the linked account.</span>")
+
+	to_chat(user, "<span class='notice'>The linked account now reports a balance of $[registered_account.account_balance].</span>")
+	qdel(I)
+
+/obj/item/stashbox/Destroy()
+	var/cached_funds = registered_account.account_balance // Funds before we reduce
+	var/refund_penalty = rand(10, 20)// Percentage to reduce by
+	var/refund = round(((refund_penalty) / 100) * (cached_funds))// Ammount to refund after it has been reduced
+	registered_account.adjust_money(-refund)
+	var/obj/item/holochip/holochip = new (src.drop_location(), refund)
+	to_chat(loc, "a credit chip falls out of [src]")
```

Comment:
cursed to_chat

Reply:
idk what else to do about that. visible_message doens't work